### PR TITLE
[DM-31082] Update cachemachine chart to match Gafaelfawr

### PR DESCRIPTION
**Pull request by rra**
Try to move towards a more standardized way of writing Helm charts
for the Science Platform by adopting the current Gafaelfawr
configuration and newer Helm chart templates and hewing more closely
to Helm best practices.  This includes:

- Using a selectorLabels template helper
- Including generic annotations on every resource
- Putting roles and role bindings in the same file with the service
  account
- Making use of Vault mandatory
- Drop configuration of the name of the Docker secret
- Avoid embedding the type of object in the name in more places
- Name files as <resource-type>-<subsystem> where relevant
- Use helm-docs for documentation
- Add the checksum of the configmap as an annotation on the
  deployment to force restarts when the configmap changes
- Simplify the ingress
- Remove configuration of the replica count since it never makes
  sense to run more than one cachemachine (they would duplicate
  each other's work)
- Drop readiness probe based on some Kubernetes best practice
  recommendations (it makes it hard to diagnose failures since they
  spontaneously disappear)

### CODE REVIEW

**Comment by rra** [comment]
Sorry for the rather large diff. I have a long-term hope of making all of our charts use consistent style and configuration options so took advantage of this change to do a lot of refactoring. Let me know if that's too much and I can do something more targeted.

File: charts/cachemachine/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+| Name | Email | Url |
+| ---- | ------ | --- |
+| cbanek |  |  |
+
+## Values
+
+| Key | Type | Default | Description |
+|-----|------|---------|-------------|
+| affinity | object | `{}` | Affinity rules for the Gafaelfawr frontend pod |
```
Comment (cbanek) [inline charts/cachemachine/README.md:17]: s/gafaelfawr/cachemachine

File: charts/cachemachine/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+| fullnameOverride | string | `""` | Override the full name for resources (includes the release name) |
+| image.pullPolicy | string | `"IfNotPresent"` | Pull policy for the cachemachine image |
+| image.repository | string | `"lsstsqre/cachemachine"` | cachemachine image to use |
+| image.tag | string | The appVersion of the chart | Tag of cachemachine image to use |
+| imagePullSecrets | list | `[]` | Secret names to use for all Docker pulls |
+| ingress.annotations | object | `{"nginx.ingress.kubernetes.io/auth-method":"GET","nginx.ingress.kubernetes.io/auth-url":"http://gafaelfawr.gafaelfawr.svc.cluster.local:8080/auth?scope=exec:admin"}` | Additional annotations to add to the ingress. |
+| ingress.enabled | bool | `false` | Whether to create an ingress |
+| ingress.host | string | None, must be set if the ingress is enabled | Hostname for the ingress |
+| nameOverride | string | `""` | Override the base name for resources |
+| nodeSelector | object | `{}` | Node selector rules for the Gafaelfawr frontend pod |
```
Comment (cbanek) [inline charts/cachemachine/README.md:28]: gafaelfawr here and couple of lines below

**Review by cbanek — APPROVED** [review]
This all looks great, thanks!  I'm fine with all this and it sounds like this will fix our config map cache problems.

**Pull request merged**